### ISSUE TITLE
Exception has occurred: KeyError in Tensor.py

### ISSUE THREAD

**Issue by iSapGod** [tensorflow/tensorflow#31658]
Hi there!
I wanna run an example from [this tutorial](https://colab.sandbox.google.com/github/tensorflow/docs/blob/r2.0rc/site/en/r2/tutorials/text/nmt_with_attention.ipynb)
Firstly I've got an error but that was fixed by adding this line:
`tf.enable_eager_execution()`
I downloaded [French - English dataset](http://www.manythings.org/anki/)
Then script was built successfully but at the end I've got another error:
```
Exception has occurred: KeyError
'desole'
  File "C:\Users\xxx\Python\Tensor.py", line 314, in <listcomp>
    inputs = [inp_lang.word_index[i] for i in sentence.split(' ')]
  File "C:\Users\xxx\Python\Tensor.py", line 314, in evaluate
    inputs = [inp_lang.word_index[i] for i in sentence.split(' ')]
  File "C:\Users\xxx\Python\Tensor.py", line 366, in translate
    result, sentence, attention_plot = evaluate(sentence)
  File "C:\Users\xxx\Python\Tensor.py", line 377, in <module>
    translate(u'Je suis désolé, mais je ne peux pas répondre tout de suite.')
```

This error occurs on this line:
`translate(u'Je suis desole, mais je ne peux pas repondre tout de suite .')`
Implementation of this function you can get from tutorial above.

**System information**
OS: latest Windows 10 build;
TensorFlow: 1.14.0;
Python: 3.7.3;
[MSC v.1916 64 bit (AMD64)];
CPU: AMD Ryzen 7 2700x;
GPU: NVidia GeForce GTX 1050 Ti.

If you need some additional info, I will provide it.
Help me please to solve this issue or tell me where I'm doing something wrong.

**Comment by ymodak** [comment]
Perhaps you can try this tutorial with TF 2.X version as mentioned in the colab.

**Comment by iSapGod** [comment]
> Perhaps you can try this tutorial with TF 2.X version as mentioned in the colab.

Thanks for reply!
I found that I had some issues with versions of TF. I fixed it but described issue is still appearing.

**Comment by ravikyram** [comment]
@iSapGod 

Looks like some words are missing in your dataset. Please, include those words and try again.Thanks!

**Comment by iSapGod** [comment]
@ravikyram 
Thanks for reply.
Yeah, I thought about it too. But then I just used sentences from dataset, and nothing changed.

**Comment by iSapGod** [comment]
@jvishnuvardhan 
This issue occurs only after model is trained and trying to get some results. All code is provided via link above (actually, code hasn't a lot of lines)

**Comment by jvishnuvardhan** [comment]
@iSapGod I ran it with T2.0 without any issues. Here is the [gist](https://colab.sandbox.google.com/gist/jvishnuvardhan/a1d51dd24e14b789285462a9324565f8/nmt_with_attention.ipynb). One modification I made was uploading `fra.txt` to colab and read it from there. Please check the gist and let me know if you see anything missing. 

When you respond or create a new TF issue, Please provide a simple standalone code (the current code (tutorial) is too long and requires lot of efforts to resolve). If you provide simple standalone code, then it is easy for localizing the issue faster. Thanks again.!

**Comment by jvishnuvardhan** [comment]
Automatically closing this out since I understand it to be resolved, but please let me know if I'm mistaken.Thanks!

**Comment by laurianekoffi** [comment]
please help me i have the same error
def evaluate(sentence):
  attention_plot = np.zeros((max_length_targ, max_length_inp))

  sentence = preprocess_sentence(sentence)

  inputs = [inp_lang.word_index[i] for i in sentence.split(' ')]
  inputs = tf.keras.preprocessing.sequence.pad_sequences([inputs],
                                                         maxlen=max_length_inp,
                                                         padding='post')
  inputs = tf.convert_to_tensor(inputs)

  result = ''

  hidden = [tf.zeros((1, units))]
  enc_out, enc_hidden = encoder(inputs, hidden)

  dec_hidden = enc_hidden
  dec_input = tf.expand_dims([targ_lang.word_index['<start>']], 0)

  for t in range(max_length_targ):
    predictions, dec_hidden, attention_weights = decoder(dec_input,
                                                         dec_hidden,
                                                         enc_out)

    # storing the attention weights to plot later on
    attention_weights = tf.reshape(attention_weights, (-1, ))
    attention_plot[t] = attention_weights.numpy()

    predicted_id = tf.argmax(predictions[0]).numpy()

    result += targ_lang.index_word[predicted_id] + ' '

    if targ_lang.index_word[predicted_id] == '<end>':
      return result, sentence, attention_plot

**Comment by laurianekoffi** [comment]
KeyError                                  Traceback (most recent call last)
[<ipython-input-41-664e526d304d>](https://localhost:8080/#) in <module>()
----> 1 translate(u'Trois')

2 frames
[<ipython-input-37-88c350b1987b>](https://localhost:8080/#) in <listcomp>(.0)
      4   sentence = preprocess_sentence(sentence)
      5 
----> 6   inputs = [inp_lang.word_index[i] for i in sentence.split(' ')]
      7   inputs = tf.keras.preprocessing.sequence.pad_sequences([inputs],
      8                                                          maxlen=max_length_inp,

KeyError: 'trois'

**Comment by jvishnuvardhan** [comment]
@laurianekoffi As closed issue was an old issue, can you please open a new issue with a simple standalone code to reproduce the issue? Thanks!
